### ISSUE TITLE
Internal error when trying to get a property on a Val:: type

### ISSUE THREAD

**Issue by masak** [masak/007#97]
$ perl6 bin/007 -e='42.foo'
    Method 'properties' not found for invocant of class 'Val::Int'
      in method property at /home/masak/ours/007/lib/_007/Runtime.pm:189
    [...]

There's simply no code path that anticipates this in the `property` method. There should be.

**Comment by andreoss** [comment]
Should it throw  more descriptive error message?  
```
$ perl6 -Ilib bin/007 -e='42.foo'
Property 'foo' not found
...
```
Or there should be a way to add properties to Val::Int? 
Like this
```
my m = Q::Literal::Int {
    value: 10,
    propertylist: Q::PropertyList { properties: [
        Q::Property {
            key: "foo",
            value : Q::Literal::Str { value: "hi"} }
    ] }
};
my i = melt(m);
say(i); # 42
say(i.foo) # hi
```

**Comment by masak** [comment]
> Should it throw more descriptive error message?

Same as for `Q` objects, `X::PropertyNotFound` is fine.

> Or there should be a way to add properties to Val::Int?

We'll get such facilities when we get a proper class system. See #32 for some discussion. For now, let's keep it strict and only allow the properties in the object constructor that are actually declared on the type.

**Comment by masak** [comment]
Addressed and fixed by ed16d3a99327ceec76dbe35606b6488dd622d846. Closing.
